apps/www/registry/new-york/ui/timeline.tsx

```diff
@@ -0,0 +1,207 @@
+    },
+  }
+)
+
+interface TimelineDotProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof timelineDotVariants> {
+  customIcon?: React.ReactNode
+}
+
+const TimelineDot = React.forwardRef<HTMLDivElement, TimelineDotProps>(
+  ({ className, status, customIcon, ...props }, ref) => (
+    <div
+      role="status"
+      className={cn("timeline-dot", timelineDotVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    >
+      <div className="radix-circle size-2.5 rounded-full" />
+      <CheckIcon className="radix-check size-3" />
+      <Cross1Icon className="radix-cross size-2.5" />
+      {customIcon}
+    </div>
+  )
+)
+TimelineDot.displayName = "TimelineDot"
+
+const timelineContentVariants = cva(
+  "row-start-2 row-end-2 pb-8 text-muted-foreground",
+  {
+    variants: {
+      side: {
+        right: "col-start-3 col-end-4 mr-auto text-left",
+        left: "col-start-1 col-end-2 ml-auto text-right",
+      },
+    },
+    defaultVariants: {
+      side: "right",
+    },
+  }
+)
+
+interface TimelineConentProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof timelineContentVariants> {}
+
+const TimelineContent = React.forwardRef<HTMLDivElement, TimelineConentProps>(
+  ({ className, side, ...props }, ref) => (
+    <div
+      className={cn(timelineContentVariants({ side }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineContent.displayName = "TimelineContent"
+
+const timelineHeadingVariants = cva(
+  "row-start-1 row-end-1 line-clamp-1 max-w-full truncate",
+  {
+    variants: {
+      side: {
+        right: "col-start-3 col-end-4 mr-auto text-left",
+        left: "col-start-1 col-end-2 ml-auto text-right",
+      },
+      variant: {
+        primary: "text-base font-medium text-primary",
+        secondary: "text-sm font-light text-muted-foreground",
+      },
+    },
+    defaultVariants: {
+      side: "right",
+      variant: "primary",
+    },
+  }
+)
+
+interface TimelineHeadingProps
+  extends React.HTMLAttributes<HTMLParagraphElement>,
+    VariantProps<typeof timelineHeadingVariants> {}
+
+const TimelineHeading = React.forwardRef<
+  HTMLParagraphElement,
+  TimelineHeadingProps
+>(({ className, side, variant, ...props }, ref) => (
+  <p
+    role="heading"
+    aria-level={variant === "primary" ? 2 : 3}
+    className={cn(timelineHeadingVariants({ side, variant }), className)}
+    ref={ref}
+    {...props}
+  />
+))
+TimelineHeading.displayName = "TimelineHeading"
+
+interface TimelineLineProps extends React.HTMLAttributes<HTMLHRElement> {
+  done?: boolean
+}
+
+const TimelineLine = React.forwardRef<HTMLHRElement, TimelineLineProps>(
+  ({ className, done = false, ...props }, ref) => {
+    return (
+      <hr
+        role="separator"
+        aria-orientation="vertical"
+        className={cn(
+          "col-start-2 col-end-3 row-start-2 row-end-2 mx-auto flex h-full min-h-16 w-0.5 justify-center rounded-full",
```

Comment:
<img width="215" alt="Screenshot 2024-09-25 at 10 26 48" src="https://github.com/user-attachments/assets/0986f6a7-178c-4273-a7a7-de0d39b4ee67">
Min height might not be desired by everybody